### PR TITLE
feat: `update --json`

### PR DESCRIPTION
**Pull request by baszalmstra**
Fixes https://github.com/prefix-dev/pixi/issues/1437

Schema:

```json
{
  "environment": {
    "linux-64": [
    {
        "name": "zstd",
        "before": {
          "version": "1.5.5",
          "build": "hfc55251_0"
        },
        "after": {
          "version": "1.5.6",
          "build": "ha6fb4c9_0"
        },
        "type": "conda"
        "explicit": false
      }
    ]
  }
}
```

@pavelzw Let me know what you think!

I changed to an array instead of a dictionary because a package changing from conda to pypi is represented as a removal followed by an insert. 

The `explicit` flag is only present if its value is `true`. 

@ruben-arts This is missing a test! Waiting for Pavel to give me the green light first.

### TODO:
- [x] Documentation, add it to the reference/cli.md
- [x] Full record spec in json output

### CODE REVIEW

File: src/cli/update.rs [hunk 1]
```diff
@@ -622,3 +638,147 @@ impl LockFileDiff {
...
+    ty: JsonPackageType,
+    #[serde(skip_serializing_if = "std::ops::Not::not")]
+    explicit: bool,
+}
+
+#[derive(Serialize, Clone)]
+pub struct JsonChange {
+    version: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    build: Option<String>,
```
Comment (ruben-arts) [inline src/cli/update.rs:657]: There are more things that can change. 

`channel` is an important one I believe, but why not print the full record, before and after? Then the user can fetch what he requires.

**Comment by pavelzw** [comment]
Sounds good! Thanks ❤️  I'll update my diff tool 🚀

File: src/cli/update.rs (hunk 1 above)
Comment (baszalmstra) [reply src/cli/update.rs:657]: I wanted to keep it in line with the human output but thats not a bad idea.

**Comment by wolfv** [comment]
I am wondering if we should add a schema version just to be safe?

**Comment by pavelzw** [comment]
For me personally, I don't really need it. I'm wondering how many people would use it, I think for most people it would be fine if it's described with an example.

**Comment by pavelzw** [comment]
Schema version sounds good 👍🏻

File: src/cli/update.rs (hunk 1 above)
Comment (pavelzw) [reply src/cli/update.rs:657]: Sounds good to me 👍🏻

**Comment by pavelzw** [comment]
https://github.com/pavelzw/pixi-diff-to-markdown is compatible with this version 👍🏻

**Comment by pavelzw** [comment]
Here are my ideas of how to use this in GitHub: https://github.com/pavelzw/pixi-pycharm/pull/21

Either as a reusable workflow hosted in the prefix-dev org or as a simple example in the docs. The latter would be a bit more flexible, so I'm leaning a bit towards the latter.

**Comment by baszalmstra** [comment]
I changed the output slightly, it now outputs the changed values as they appear in the lock-file. 

```json
    {
        "name": "typing_extensions",
        "before": {
          "version": "4.11.0",
          "url": "https://conda.anaconda.org/conda-forge/noarch/typing_extensions-4.11.0-pyha770c72_0.conda",
          "sha256": "a7e8714d14f854058e971a6ed44f18cc37cc685f98ddefb2e6b7899a0cc4d1a2",
          "md5": "6ef2fc37559256cf682d8b3375e89b80",
          "size": 37583,
          "timestamp": 1712330089194
        },
        "after": {
          "timestamp": 1717287863652,
          "size": 39706,
          "version": "4.12.1",
          "url": "https://conda.anaconda.org/conda-forge/noarch/typing_extensions-4.12.1-pyha770c72_0.conda",
          "sha256": "c50d61fe29cd2752943358037ee1107a60b44b8d32c464d18308d668b6494573",
          "md5": "26d7ee34132362115093717c706c384c"
        },
        "type": "conda"
      },
```

**Comment by baszalmstra** [comment]
The schema is now:

```json
{
    "version": 1,
    "environments": {
    	"<name>": {
			"<platform>": [
			    {
					"name": "<package_name>",
					"type": "conda"|"pypi"
					"before": {...} | null,
					"after": {...} | null,
					"explicit": bool,
				}
			]
        }
    }
}
```

**Review by ruben-arts — APPROVED** [review]
Read it, Ran it, Loved it, Liked it, Merged it!

**Pull request merged**